### ISSUE TITLE
Unnecessary complexity in repo.sls states re permissions

### ISSUE THREAD

**Issue by smitty42** [terminal-labs/sample-states#34]
<a href="https://github.com/nixjdm"><img src="https://avatars1.githubusercontent.com/u/3431410?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [nixjdm](https://github.com/nixjdm)**
_Saturday Oct 28, 2017 at 21:37 GMT_
_Originally opened as https://github.com/terminal-labs/rambo/issues/122_

----

Both the hg and git cloning states perform the clone as root, and then have a separate state setting normal permissions. I presume we can just clone it as a right user and not need the second state.

https://github.com/terminal-labs/rambo/blob/master/rambo/salt_resources/states/hg/repo.sls#L5-L14
https://github.com/terminal-labs/rambo/blob/master/rambo/salt_resources/states/git/repo.sls#L5-L14

**Comment by smitty42** [comment]
<a href="https://github.com/nixjdm"><img src="https://avatars1.githubusercontent.com/u/3431410?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [nixjdm](https://github.com/nixjdm)**
_Wednesday Nov 22, 2017 at 17:46 GMT_

----

@alan-cugler A good first salt issue

**Comment by smitty42** [comment]
<a href="https://github.com/nixjdm"><img src="https://avatars1.githubusercontent.com/u/3431410?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [nixjdm](https://github.com/nixjdm)**
_Thursday Nov 23, 2017 at 00:11 GMT_

----

This is not as easy as I thought. The complexity is that the deescalated user needs the appropriate keys to clone from a remote server. The root user has those keys, but the deescalated user doesn't. I'm making a new ticket for that though. #188 should be done before this ticket.
